Sources/Sentry/SentryProfiler.mm

```diff
@@ -0,0 +1,228 @@
+                    frame[@"instruction_addr"] = sentry_formatHexAddress(@(backtrace.addresses[i]));
+#    if defined(DEBUG)
+                    frame[@"function"] = [NSString stringWithUTF8String:symbols[i]];
+#    endif
+                    [frames addObject:frame];
+                }
+
+                const auto sample = [NSMutableDictionary<NSString *, id> dictionary];
+                sample[@"frames"] = frames;
+                sample[@"relative_timestamp_ns"] =
+                    [[strongSelf getElapsedDuration:backtrace.uptimeNs] stringValue];
+                sample[@"thread_id"] = threadID;
+                [samples addObject:sample];
+            },
+            100 /** Sample 100 times per second */);
+        _profiler->startSampling();
+    }
+}
+
+- (void)stop
+{
+    @synchronized(self) {
+        _profiler->stopSampling();
+    }
+}
+
+- (SentryEnvelopeItem *)buildEnvelopeItemForTransaction:(SentryTransaction *)transaction
+{
+    NSMutableDictionary<NSString *, id> *profile = nil;
+    @synchronized(self) {
+        profile = [_profile mutableCopy];
+    }
+    const auto debugImages = [NSMutableArray<NSDictionary<NSString *, id> *> new];
+    const auto debugMeta = [_debugImageProvider getDebugImages];
+    for (SentryDebugMeta *debugImage in debugMeta) {
+        [debugImages addObject:[debugImage serialize]];
+    }
+    if (debugImages.count > 0) {
+        profile[@"debug_meta"] = @{ @"images" : debugImages };
+    }
+
+    profile[@"device_locale"] = NSLocale.currentLocale.localeIdentifier;
+    profile[@"device_manufacturer"] = @"Apple";
+    const auto model = getDeviceModel();
+    profile[@"device_model"] = model;
+    profile[@"device_os_build_number"] = getOSBuildNumber();
+#    if SENTRY_HAS_UIKIT
+    profile[@"device_os_name"] = UIDevice.currentDevice.systemName;
+    profile[@"device_os_version"] = UIDevice.currentDevice.systemVersion;
+#    endif
+    profile[@"device_is_emulator"] = @(isSimulatorBuild());
+    profile[@"device_physical_memory_bytes"] =
+        [@(NSProcessInfo.processInfo.physicalMemory) stringValue];
+    profile[@"environment"] = transaction.environment;
+    profile[@"platform"] = transaction.platform;
+    profile[@"transaction_id"] = transaction.eventId.sentryIdString;
+    profile[@"trace_id"] = transaction.trace.context.traceId.sentryIdString;
+    profile[@"profile_id"] = [[SentryId alloc] init].sentryIdString;
+    profile[@"transaction_name"] = transaction.transaction;
+    profile[@"duration_ns"] = [[self getElapsedDuration:getReferenceTimestamp()] stringValue];
+
+    const auto bundle = NSBundle.mainBundle;
+    profile[@"version_code"] = [bundle objectForInfoDictionaryKey:(NSString *)kCFBundleVersionKey];
+    profile[@"version_name"] = [bundle objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
+
+    NSError *error = nil;
+    const auto JSONData = [NSJSONSerialization dataWithJSONObject:profile options:0 error:&error];
```

Comment:
l: We have a `SentrySerialization` class. Can we use that here?

I don't know any specifics on `SentrySerialization` and why this class was created, since this is from before I joined Sentry. Just wondering if we should use it when we need to serialize json.

@bruno-garcia @philipphofmann Any thoughts?